In [12]:
import pandas as pd 

reviews = pd.read_csv('Datasets\winemag\winemag-data-130k-v2.csv', index_col=0)
reviews.head()

country                                        description  \
0     Italy  Aromas include tropical fruit, broom, brimston...   
1  Portugal  This is ripe and fruity, a wine that is smooth...   
2        US  Tart and snappy, the flavors of lime flesh and...   
3        US  Pineapple rind, lemon pith and orange blossom ...   
4        US  Much like the regular bottling from 2012, this...   

                          designation  points  price           province  \
0                        Vulkà Bianco      87    NaN  Sicily & Sardinia   
1                            Avidagos      87   15.0              Douro   
2                                 NaN      87   14.0             Oregon   
3                Reserve Late Harvest      87   13.0           Michigan   
4  Vintner's Reserve Wild Child Block      87   65.0             Oregon   

              region_1           region_2         taster_name  \
0                 Etna                NaN       Kerin O’Keefe   
1                  NaN                NaN          Roger Voss   
2    Willamette Valley  Willamette Valley        Paul Gregutt   
3  Lake Michigan Shore                NaN  Alexander Peartree   
4    Willamette Valley  Willamette Valley        Paul Gregutt   

  taster_twitter_handle                                              title  \
0          @kerinokeefe                  Nicosia 2013 Vulkà Bianco  (Etna)   
1            @vossroger      Quinta dos Avidagos 2011 Avidagos Red (Douro)   
2           @paulgwine       Rainstorm 2013 Pinot Gris (Willamette Valley)   
3                   NaN  St. Julian 2013 Reserve Late Harvest Riesling ...   
4           @paulgwine   Sweet Cheeks 2012 Vintner's Reserve Wild Child...   

          variety               winery  
0     White Blend              Nicosia  
1  Portuguese Red  Quinta dos Avidagos  
2      Pinot Gris            Rainstorm  
3        Riesling           St. Julian  
4      Pinot Noir         Sweet Cheeks

1.
What is the data type of the points column in the dataset?

In [ ]:
reviews.points.dtypes

2.
Create a Series from entries in the points column, but convert the entries to strings. Hint: strings are str in native Python.

In [ ]:
point_strings = reviews.points.astype(str)
reviews.points.dtypes

In [ ]:
point_strings

In [ ]:
type(point_strings)

3.
Sometimes the price column is null. How many reviews in the dataset are missing a price?

In [10]:
n_missing = reviews.isnull().sum()
n_missing

country                     63
description                  0
designation              37465
points                       0
price                     8996
province                    63
region_1                 21247
region_2                 79460
taster_name              26244
taster_twitter_handle    31213
title                        0
variety                      1
winery                       0
dtype: int64

In [11]:
n_missing_prices =reviews.price.isnull().sum()
n_missing_prices

8996

4.
What are the most common wine-producing regions? Create a Series counting the number of times each value occurs in the region_1 field. This field is often missing data, so replace missing values with Unknown. Sort in descending order. Your output should look something like this:

Unknown                    21247
Napa Valley                 4480
                           ...  
Bardolino Superiore            1
Primitivo del Tarantino        1
Name: region_1, Length: 1230, dtype: int64

In [36]:
#groupby doest group NA values 
reviews_per_region = reviews.groupby("region_1").size().sort_values(ascending=False).isna().sum()
reviews_per_region

0

In [35]:

reviews_per_region = reviews.region_1.fillna("Unknow").value_counts().sort_values(ascending=False)
reviews_per_region.loc["Unknow"]

21247

In [37]:
reviews.dropna(axis=0,subset=["region_1"], inplace=True)

In [38]:
reviews.region_1.isna().sum()

0